src/Event/MenuEvent.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Contao\CoreBundle\Event;
+
+use Knp\Menu\ItemInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+class MenuEvent extends Event
+{
+    private $tree;
+
+    public function __construct(ItemInterface $tree)
```

Comment:
The event should also contain the menu factory, so it does not need to be injected into the listener.